File: matlab_design/test_rms.m
%% Root Mean Square Test
% this is a very minimal test for comparing the c routines for the
% rms calculation.
% The values were set the same in the c test and the same answer of an rms
% of 0.5339 was achieved by this script, the c routine, and by hand.
%   

% Jacob Allenwood
% October 26, 2015


%% RMS Calculation

block_size = 10;
square = 0;
x = [0.0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9];
y = 0;

% go through entire input
for j=1:block_size    
    % sum the squares for the rms calc
    square = square + x(j)^2;  
end

% calc rms value
y = sqrt(square / block_size);
    

File: main/test_rms.c
/**
 * @file effect_main.c
 *
 * @author Jacob Allenwood
 * @date September 1, 2015
 *
 * @brief This file contains the main program to run the selected GAPE effect
 * 
 */

// include files -------------------------------------------------------
#include <stdlib.h>
#include <stdio.h>

#include "calc_rms.h"

// ---------------------------------------------------------------------




// this is a very minimal test, but I did the calculation by hand, 
 // the matlab test script, and this routine, and found the same answer of 
 // 0.5339
 // More extensive testing, for multiple blocks should probably be implemented.

int main(int argc, char const *argv[]) {


	int block_size, i, j;
	block_size = 10;
	float input[10] = {0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9};


	// setup rms struct
	RMS_T * V = init_rms(10, block_size);

	calc_rms(V, input);

	// for(i = 0; i < block_size; i++) {
	// 	printf("%f,", V->output[i]);
	// }

	printf("%f", V->output[block_size - 1]);


}




